test/Dialect/SV/hw-extract-test-code.mlir

```diff
@@ -6,19 +6,19 @@
 // CHECK-NOT: attributes
 // CHECK-NEXT: hw.module.extern @foo_assert
 // CHECK-NOT: attributes
-// CHECK: hw.module @issue1246_assert(in %clock : i1) attributes {comment = "VCS coverage exclude_file", output_file = #hw.output_file<"dir3{{/|\\\\}}", excludeFromFileList, includeReplicatedOps>}
+// CHECK: hw.module @issue1246_assert(in %{{[^ ]+}} : i1) attributes {comment = "VCS coverage exclude_file", output_file = #hw.output_file<"dir3{{/|\\\\}}", excludeFromFileList, includeReplicatedOps>}
```

Comment:
What are the new names generated here? I wouldn't think that the non-anonymous names would change?

Reply:
The old names are also changed, a port index suffix is added. This is required to hide a bug in `LegalizeNames` , port renaming is not reflected on the instance ops. So, if the port names conflict, then the final output mlir has inconsistent port names on module and instance op.

Reply:
But what exactly are the new names? Is the clock not called `clock` anymore?

Reply:
Its called `clock0`, which is suffixed by the port index.

Reply:
So this will add index suffixes to the names of all ports for ETC-generated modules? If so that is not desirable.

Reply:
I fixed the logic, to unique names only on conflict.